.github/workflows/build.yml

```diff
@@ -0,0 +1,33 @@
+name: Build
+on:
+  pull_request:
+
+env:
+  SONAR_ORG: flexbase
+  SONAR_PROJECT_KEY: flexbase-eng_http-client-middleware
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # Disabling shallow clone is recommended for improving relevancy of reporting
+    - uses: actions/setup-node@v3
+    - name: yarn install
+      run: yarn install
+    - name: Build
+      run: yarn build
+    - name: Lint
+      run: yarn lint
+    - name: Test
+      run: yarn test
+    - name: SonarCloud Scan
```

Comment:
if yarn is the standard for most repos, we can move this to a shared workflow in an `internal` repo, which can be referenced by other repositories to run these jobs.

im not sure about the shared secrets portion, but then i believe we could also just keep sonarqube secrets we want used in the shared repo as well.

Reply:
secrets are org wide so no issue there. There are a few ways to do this and would like to go over them. The ones I can think of is write our own GAs that wrap this, dockerizer with a base image that does this stuff, ?